### ISSUE TITLE
[Bug]: Legacy MSBuild projects sometimes fail to build

### ISSUE THREAD

**Issue by timothylcooke** [reactiveui/ReactiveUI.SourceGenerators#144]
### Describe the bug 🐞

By adding a Window with a namespace definition in an SDK-style .NET Framework WPF application, we can essentially prevent SourceGenerators from generating code. The IDE makes it look like everything is fine: you can navigate to the generated source file for properties generated by the `[ReactiveUI.SourceGenerators.ReactiveAttribute]`, but the project fails to compile:

```
error CS0234: The type or namespace name 'SourceGenerators' does not exist in the namespace 'ReactiveUI' (are you missing an assembly reference?)
```

### Step to reproduce

[This project](https://github.com/timothylcooke/SourceGeneratorIssue) is a minimal reproduction. [Initial commit b27e435](https://github.com/timothylcooke/SourceGeneratorIssue/commit/b27e435d0a6a31bd1a792b813659e72b54eadf62) builds just fine in both a legacy and sdk-style project. [a845b4a](https://github.com/timothylcooke/SourceGeneratorIssue/commit/a845b4adacfd8e29bb126c3580046c00476e4473) adds an `xmlns:local` definition to XAML. Once that namespace is added, the legacy application no longer compiles (though the SDK-style project continues to compile).

### Reproduction repository

https://github.com/timothylcooke/SourceGeneratorIssue

### Expected behavior

The project should compile

### Operating system

Windows 11 23H2

### Version

Visual Studio 17.12.3

### ReactiveUI Version

2.0.17

### Additional information ℹ️

It's unclear if this is even your bug, or if it's an issue in msbuild.

I have a legacy WPF application deployed with ClickOnce, so I can't switch to an SDK-style project without a whole lot of headache. While `ReactiveUI.Fody` works fine, I'd like to start using `ReactiveUI.SourceGenerators`.

Adding local namespaces in XAML is a very common thing in WPF that I can't just avoid doing. The best workaround I have is to move any VM classes to a library.

If you remove all references to `ReactiveAttribute` to get the project to compile, you can access the type via reflection: `var reactiveAttributeTypeViaReflection = GetType().Assembly.GetType("ReactiveUI.SourceGenerators.ReactiveAttribute");` So, it seems like it's an issue with the order of build events: it seems that the generated files are being added _after_ the static files are built, so any references from static files to generated types fail to compile because the types haven't been generated yet.

**Comment by glennawatson** [comment]
I'm a bit behind on a lot of maintenance stuff in regards to this project and @ChrisPulman is out til christmas. 

There can be issues with source generator relying on source generators. I know they moved some of the xaml into a generated based context, not sure if they are using source generators to do.

**Comment by glennawatson** [comment]
Usually the way I've solved source generator relying on source generator issues is to move into separate projects the source code so only one generator happens at any one time.

**Comment by ChrisPulman** [comment]
<img width="964" alt="image" src="https://github.com/user-attachments/assets/f5ca8b59-c0b3-4e86-b5de-077c45befd74" />
From the looks of things, the ReactiveUI Source Generator is generating the files correctly, therefore my guess is that it is an issue with MS-Build and the way it works with source generators, Fody injects IL modifications after the build is complete hence this will work. The most viable work around may be to use a netstandard2.0 library project for ViewModels and Source Generated elements and then keep the Views and ViewModels separated. I have tested this option and it does work; but I guess it depends on your individual project.

**Comment by timothylcooke** [comment]
Yeah, @ChrisPulman. That's exactly what I've been doing. Thanks for looking into this. I gather that there's nothing to do here except close this issue and possibly open an issue with msbuild?

**Comment by timothylcooke** [comment]
> From the looks of things, the ReactiveUI Source Generator is generating the files correctly

The Source Generator is definitely working. Generated types exist at runtime. It's just that they can _only_ be referenced via reflection. Static references cause build failures. Very strange behavior.

**Comment by glennawatson** [comment]
Probably try the latest source gen we just published, see if it working, then chase up with msbuild if that fails.

**Comment by ChrisPulman** [comment]
Version 2.1.8 has been released, there may have been some issues with a version of Visual Studio, if you still feel that there is something wrong with the ReactiveUI Source Generators please reopen this or create a new issue with any updates, thank you.
